deep_quoridor/src/agents/simple.py

```diff
@@ -1,61 +1,175 @@
 import random
+from dataclasses import dataclass, fields
+from typing import Optional
 
-from quoridor import ActionEncoder, Player, Quoridor, construct_game_from_observation
-from quoridor_env import StepRewardCalculator
+import numpy as np
+from quoridor import Action, ActionEncoder, Player, Quoridor, WallAction, construct_game_from_observation
+from utils import SubargsBase
 
 from agents.core import Agent
 
+# We use a large reward to encourage the agent to win the game, but we can't use infinity
+# because then multiplying by a discount factor won't decrease the reward.
+WINNING_REWARD = 1e6
 
-def sample_random_action_sequence(game: Quoridor, player: Player, opponent: Player, max_path_length: int):
+
+@dataclass
+class SimpleParams(SubargsBase):
+    nick: Optional[str] = None
+
+    # How many moves to look ahead in the minimax algorithm.
+    max_depth: int = 2
+
+    # How many actions to consider at each minimax stage.
+    branching_factor: int = 20
+
+    # We use a Guassian distribution centered around each player's position to sample wall actions.
+    # Sigma is the standard deviation of this distribution; a smaller sigma means the agent is more
+    # likely to place walls near themselves or their opponent, rather than somewhere else on the board.
+    wall_sigma: float = 0.5
```

Comment:
I think that usually you want to put walls around the opponent, except at the beginning when you're marching towards each other and you don't want to block your own path.  Maybe using 2 different wall sigmas so you look less on your side could trim unnecessary actions?

Reply:
I think you may be right; I'd like to explore that idea in a future PR. Having different standard deviations around each player could work, but it might be be better to have a scaling factor on the value instead (since decreasing or increasing the standard devation of a gaussian doesn't actually change the total probability of it)